packages/react-aria-components/test/Table.test.js

```diff
@@ -2167,6 +2189,26 @@ describe('Table', () => {
       expect(rows[3]).toHaveTextContent('Trailblazer');
       expect(rows[4]).toHaveTextContent('RatSat');
     });
+
+    it('should not render excessively in React Suspense with transitions', async () => {
+      // Only supported in React 19.
+      if (!React.use) {
+        return;
+      }
+
+      jest.useRealTimers();
+
+      let tree = await act(() => render(<TableWithSuspense reactTransition />));
+
+      await act(() => stories.makePromise([]));
+
+      expect(mockCollectionUpdate).toHaveBeenCalledTimes(1);
+
+      let button = tree.getByRole('button');
+      await act(() => button.click());
+
+      expect(mockCollectionUpdate).toHaveBeenCalledTimes(2);
```

Comment:
Nice, I think this is a better representation of what we care about regressing in the future. Thanks!